airflow/contrib/sensors/qubole_sensor.py

```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+from future import standard_library
+standard_library.install_aliases()
+from builtins import str
+import logging
+
+from airflow.operators.sensors import BaseSensorOperator
+from airflow.hooks.base_hook import BaseHook
+from airflow.utils import apply_defaults
+from airflow.exceptions import AirflowException
+
+from qds_sdk.qubole import Qubole
```

Comment:
When I try to run the tests:
```
root@0b98ce2a6b67:/airflow/airflow# nosetests  tests/contrib/sensors/test_qubole_sensor.py
E
======================================================================
ERROR: Failure: ModuleNotFoundError (No module named 'qds_sdk')
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/nose/failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "/usr/local/lib/python3.6/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/local/lib/python3.6/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/local/lib/python3.6/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/usr/local/lib/python3.6/imp.py", line 234, in load_module
    return load_source(name, filename, file)
  File "/usr/local/lib/python3.6/imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 675, in _load
  File "<frozen importlib._bootstrap>", line 655, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 205, in _call_with_frames_removed
  File "/airflow/airflow/tests/contrib/sensors/test_qubole_sensor.py", line 25, in <module>
    from airflow.contrib.sensors.qubole_sensor import QuboleFileSensor, QubolePartitionSensor
  File "/airflow/airflow/airflow/contrib/sensors/qubole_sensor.py", line 26, in <module>
    from qds_sdk.qubole import Qubole
ModuleNotFoundError: No module named 'qds_sdk'
```
Can you make sure that the `qds_sdk` is installed?

Reply:
@Fokko thanks for the review. Will fix all the logger related suggestions, but I am not sure how to fix the installation of `qds_sdk` for running tests locally. AFAIK this module has to go in `setup.py` itself, which might not be required/preferred by many. These test are running fine in travis.

Reply:
Good morning @msumit, this is a common misconception, the Travis does not test any files in the `/contrib/` folder. I hope this changes in the future, because of lot of work is being done here.

About testing, for me, the best way of testing it is in a clean Docker container:
```
Fokkos-MBP:Desktop fokkodriesprong$ docker run --rm -t -i  python bash
root@81d6bd3e4a16:/# git clone https://github.com/msumit/airflow.git /airflow/
Cloning into '/airflow'...
remote: Counting objects: 23699, done.
remote: Total 23699 (delta 0), reused 0 (delta 0), pack-reused 23698
Receiving objects: 100% (23699/23699), 12.67 MiB | 199.00 KiB/s, done.
Resolving deltas: 100% (16751/16751), done.
Checking connectivity... done.
root@81d6bd3e4a16:/# cd /airflow/
root@81d6bd3e4a16:/airflow# git checkout AIRFLOW-1315
Branch AIRFLOW-1315 set up to track remote branch AIRFLOW-1315 from origin.
Switched to a new branch 'AIRFLOW-1315'
root@ae1f3e7af284:/airflow# pip install -e .[devel]
root@ae1f3e7af284:/airflow# pip install unicodecsv psycopg2 hive-thrift-py
... Lot of output ...
root@ae1f3e7af284:/airflow# airflow initdb
root@d11ac6a39103:/airflow# nosetests  tests/contrib/sensors/test_qubole_sensor.py
E
======================================================================
ERROR: Failure: ImportError (No module named qds_sdk.qubole)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/site-packages/nose/loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "/usr/local/lib/python2.7/site-packages/nose/importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "/usr/local/lib/python2.7/site-packages/nose/importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "/airflow/tests/contrib/sensors/test_qubole_sensor.py", line 25, in <module>
    from airflow.contrib.sensors.qubole_sensor import QuboleFileSensor, QubolePartitionSensor
  File "/airflow/airflow/contrib/sensors/qubole_sensor.py", line 24, in <module>
    from qds_sdk.qubole import Qubole
ImportError: No module named qds_sdk.qubole
-------------------- >> begin captured logging << --------------------
root: INFO: Reading the config from /root/airflow/airflow.cfg
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 1 test in 0.001s

FAILED (errors=1)
```
It would be awesome if when installing the `dev` environment using pip, the `qds_sdk` also will be installed. As you might notice, there are already a few packages that are nog being installed (`unicodecsv` `psycopg2` `hive-thrift-py`), but are required to run the tests. Hope this helps.

Reply:
Hi @Fokko thanks for the docker suggestion, but I am still having a hard time in digesting the fact that Travis didn't run contrib tests. As I had failures in Travis initially, which I resolved. Also in the travis logs I can see [success] messages against the tests I've written. Is it some new change to not run tests from contrib? as I can below from last travis run. 

`[success] 0.00% tests.contrib.sensors.test_qubole_sensor.QuboleSensorTest.test_file_sensore: [32m0.0157s[0m`
`[success] 0.00% tests.contrib.sensors.test_qubole_sensor.QuboleSensorTest.test_partition_sensor: [32m0.0153s[0m`
`[success] 0.00% tests.contrib.sensors.test_qubole_sensor.QuboleSensorTest.test_partition_sensor_error: [32m0.0134s[0m`